ui/src/components/InOutSwitch/InOutSwitch.jsx

```diff
@@ -1,110 +1,55 @@
-/* eslint-disable react/jsx-handler-names */
 import React from 'react';
 import { Badge, Button, OverlayTrigger, Popover } from 'react-bootstrap';
-import './style.css';
+import styles from './inOutStyle.module.css';
 
 export default class InOutSwitch extends React.Component {
   constructor(props) {
     super(props);
-    this.setOff = this.setOff.bind(this);
-    this.setOn = this.setOn.bind(this);
-    this.onLinkRightClick = this.onLinkRightClick.bind(this);
-    this.onOptionsRightClick = this.onOptionsRightClick.bind(this);
-
-    this.showLabelOvelay = this.showLabelOvelay.bind(this);
-    this.showValueOvelay = this.showValueOvelay.bind(this);
-
-    this.state = {
-      showLabelOvelay: false,
-      showValueOvelay: false,
-    };
-  }
-
-  showLabelOvelay(value) {
-    this.setState({
-      showLabelOvelay: value,
-    });
-  }
-
-  showValueOvelay(value) {
-    this.setState({
-      showValueOvelay: value,
-    });
-  }
-
-  /**
-   * Set Ovelay visibility to False
-   *
-   * @param {MouseEvent} e
-   */
-  onKeyDown(e) {
-    // eslint-disable-next-line sonarjs/no-small-switch
-    switch (e.key) {
-      case 'Escape': {
-        this.showValueOvelay(false);
-        this.showLabelOvelay(false);
-
-        break;
-      }
-      // No default
-    }
-  }
-
-  onLinkRightClick(e) {
-    // this.overlay.handleToggle();
-    this.showValueOvelay(!this.state.showValueOvelay);
-    e.preventDefault();
-  }
-
-  onOptionsRightClick(e) {
-    // this.optionsOverlay.handleToggle();
-    this.showLabelOvelay(!this.state.showLabelOvelay);
-    e.preventDefault();
+    this.handleSetOff = this.handleSetOff.bind(this);
+    this.handleSetOn = this.handleSetOn.bind(this);
   }
 
-  setOff() {
+  handleSetOff() {
     if (this.props.onSave !== undefined) {
-      this.props.onSave(this.props.pkey, this.props.offText);
+      if (this.props.offValue === undefined) {
+        this.props.onSave(this.props.pkey, this.props.offText);
+      } else if (this.props.pkey === undefined) {
+        this.props.onSave(this.props.offValue);
+      } else {
+        this.props.onSave(this.props.pkey, this.props.offValue);
+      }
     }
-
-    // this.overlay.hide();
-    this.showValueOvelay(false);
   }
 
-  setOn() {
+  handleSetOn() {
```

Comment:
I feel like it would be simpler to have a single event handler with an extra line at the top that, given my previous comments, would look something like:

```js
const newValue = this.isOn ? this.props.offValue : this.props.onValue;
```